nautobot/dcim/tests/test_natural_ordering.py

```diff
@@ -34,35 +34,35 @@ def test_interface_ordering_numeric(self):
             "0",
             "0.0",
             "0.1",
-            "0.2",
-            "0.10",
-            "0.100",
             "0:1",
             "0:1.0",
             "0:1.1",
             "0:1.2",
             "0:1.10",
+            "0.2",
             "0:2",
             "0:2.0",
             "0:2.1",
             "0:2.2",
             "0:2.10",
+            "0.10",
+            "0.100",
```

Comment:
At first glance this seems bad since is apparently breaking the natural ordering asserted by `CollateAsChar`... but then when you see the minor changes it actually looks more like a bug fix. 1.10 certainly should be coming after 1.2 for example!

Reply:
@lampwins Can you please take a look-see at this one?

Reply:
I am not following this one. This looks like maybe an oversight in `.` vs `:`? I.e The original ordering has `0.2` before `0:1` which would be correct.

Reply:
I'm not clear on what the semantics are supposed to be here. Consider the case of:
`192.1, 192:1, 193.1, 193:1` with CollateAsChar. 

I proposed we change this to all IPv4 in numeric order, then all IPv6 in numeric order

Reply:
This is ordering of interfaces, subinterfaces, channelized interfaces, etc., not address ordering. Consider:

Ethernet1/1  (physical interface)
Ethernet1/1.100 (subinterface 100 of Ethernet1/1)
Ethernet1/1:2 (channelized or virtual interface 2 on Ethernet1/1)

From a UI standpoint it would be preferable for all `x:yyy` values to come either before or after all `x.zzz` values since they have different semantic meanings, rather than interleaving as `x.1`, `x:1`, `x.2`, `x:2`.

Reply:
FWIW this goes all the way back to https://github.com/netbox-community/netbox/issues/9 (!) Though https://github.com/netbox-community/netbox/issues/2165 and https://github.com/netbox-community/netbox/issues/3097 are probably more relevant.